compiler/rustc_target/src/abi/mod.rs

```diff
@@ -619,7 +619,7 @@ pub struct Scalar {
 
 impl Scalar {
     pub fn is_bool(&self) -> bool {
-        if let Int(I8, _) = self.value { self.valid_range == (0..=1) } else { false }
+        if let Int(I8, false) = self.value { self.valid_range == (0..=1) } else { false }
```

Comment:
This could be changed to use `matches!`. It might be clearer:

```suggestion
        matches!(self.value, Int(I8, false)) && self.valid_range == (0..=1)
```